sn_cli/src/subcommands/files/mod.rs

```diff
@@ -159,27 +159,38 @@ async fn upload_files(
         bail!("The wallet is empty. Cannot upload any files! Please transfer some funds into the wallet");
     }
     let mut chunk_manager = ChunkManager::new(root_dir);
-    chunk_manager.chunk_path(&files_path)?;
+    chunk_manager.chunk_path(&files_path, true)?;
 
-    // Return early if we have no chunks to upload
+    // Return early if we already uploaded them
+    let chunks_to_upload;
     if chunk_manager.is_chunks_empty() {
-        println!("All files were already uploaded and verified");
-        println!("**************************************");
-        println!("*          Uploaded Files            *");
-        println!("**************************************");
-        for (file_name, addr) in chunk_manager.verified_files() {
-            if let Some(file_name) = file_name.to_str() {
-                println!("\"{file_name}\" {addr:x}");
-                info!("Uploaded {file_name} to {addr:x}");
-            } else {
-                println!("\"{file_name:?}\" {addr:x}");
-                info!("Uploaded {file_name:?} to {addr:x}");
+        // make sure we don't have any failed chunks in those
+        let chunks = chunk_manager.already_put_chunks(&files_path)?;
```

Comment:
I think this would `create chunks in the artifacts` dir and if they're not marked as completed when `failed_chunks is not empty (L172)`, then during the next run, we'd get resume the chunks at line 162 and we'll try to upload them again (L166 returns false).

Reply:
So should we make ` chunk_manager.already_put_chunks(&files_path)?;` create the `chunks` to a temp dir for verification?

Reply:
Nice catch! Marked the non-failed chunks as completed just below. And if they fail, it's reasonable to keep them around for next time!